crates/air-lib/air-parser/src/parser/lexer/call_variable_parser.rs

```diff
@@ -387,12 +413,27 @@ impl MetTag {
     fn from_tag(tag: char) -> Self {
         match tag {
             '$' => Self::Stream,
-            '#' => Self::CanonStream,
+            '#' => Self::Canon,
             '%' => Self::StreamMap,
             _ => Self::None,
         }
     }
 
+    fn update_type(&self, tag: char) -> Self {
+        match tag {
+            '$' => Self::CanonStream,
+            _ => self.to_owned(),
+        }
+    }
+
+    fn is_canon(&self) -> bool {
+        matches!(self, Self::Canon)
```

Comment:
If you derive `PartialEq` on `MetTag`, you may simply write `*self == Self::Canon`.

It worth deriving `Eq` too, as it OK for this type.

Reply:
What is a benefit comparing with the current code?

Reply:
Simple equality seems to be more straightforward.

Reply:
I tend to stick to whatever is used now and see no real benefit changing this to equality b/c it is internally the same.

Reply:
Code style is not about internally at all.

Reply:
IMHO it is unwise to use Eq here given that is_tag() uses !matches and I am not changing is_tag b/c it is outside the scope of this patch. You suggest to break the style from my point of view.

Reply:
Using a PR for small style/idiomatic changes is OK, otherwise they will never be changed.

`matches!` should be used for types that cannot be compared for some reason ("I haven't defined PartialEq just because" is not the reason).  This enum is essentially simple C-style enum.